### ISSUE TITLE
Some suggestions for the useForm hook

### ISSUE THREAD

**Issue by mediafreakch** [llauderesv/react-form-validation#1]
Hey there,

Great work! I like the simplicity of this hook! 

There are two modifications I would like to point out which were helpful for me:

1.  Use a validation function instead of a regex (or allow both):
    
     Validation Rules:
     ```
     const rules = {
       email: {
          required: true,
          validator: {
            func: value => !alreadyTaken(value),
            error: 'E-Mail address already taken'
          }
       }
     }
     ```

      Implementation (inside `useForm.js`):
      ```
         if (isObject(validationSchema[name].validator)) {
            // Test your defined validation function
            if (value && !validationSchema[name].validator.func(value)) {
               error = validationSchema[name].validator.error
            }
         }
      ```
2. It would have been useful to have a `clear()` method exported by the hook. I added:

    ```
    const clear = useCallback(() => {
      setState(stateSchema)
    }, [])

   return { handleOnChange, handleOnSubmit, state, disable, clear }
   ```

Also, you don't happen to have Typescript types laying around? 😄

**Comment by llauderesv** [comment]
@mediafreakch This is cool kindly send me PR and I'll merge your solution in this repository. I'm planning also to make a big changes in this useForm custom hook. We will keep you posted.

**Comment by llauderesv** [comment]
@mediafreakch See my fixed here [#302233d](https://github.com/llauderesv/react-form-validation/commit/3384bde679e40bff02970feedbcf874255893bf0). Appreciated your suggestion here hope you could also contribute also 🙂
